Cesium.CodeGen/Ir/Expressions/CompoundObjectInitializationExpression.cs

```diff
@@ -51,26 +55,42 @@ public void EmitTo(IEmitScope scope)
                 if (expr == null)
                     throw new CompilationException($"Retrieved null initializer!");
 
+                
+                if (_prefixAction is not null)
+                {
+                    _prefixAction();
+                    instructions.Add(Instruction.Create(OpCodes.Ldflda, _typeDef));
+                }
                 if (i != 0)
                 {
                     instructions.Add(Instruction.Create(OpCodes.Ldc_I4, i));
                     instructions.Add(Instruction.Create(OpCodes.Sizeof, element)); // size = sizeof(array element)
                     instructions.Add(Instruction.Create(OpCodes.Mul)); // offset = id * size
+                    instructions.Add(Instruction.Create(OpCodes.Add)); // offset = id * size
                 }
                 expr.EmitTo(scope);
                 instructions.Add(GetWriteInstruction(element.MetadataType));
             }
             return;
         }
 
+        _prefixAction?.Invoke();
         var newobj = new VariableDefinition(typeDef);
         scope.Method.Body.Variables.Add(newobj);
 
         instructions.Add(Instruction.Create(OpCodes.Ldloca, newobj));
         instructions.Add(Instruction.Create(OpCodes.Initobj, typeDef));
 
         if (initializers.Length == 0) // zero init like SomeType name = { };
+        {
+            instructions.Add(Instruction.Create(OpCodes.Ldloc, newobj)); // push new object
+            if (_prefixAction is not null)
+            {
+                _prefixAction();
```

Comment:
Is it okay that `_prefixAction` gets called twice? Let's maybe write some comment about its purpose and possible behavior?